silkworm/core/execution/evm.cpp

```diff
@@ -141,13 +141,13 @@ evmc::Result EVM::create(const evmc_message& message) noexcept {
     state_.add_to_balance(contract_addr, value);
 
     const evmc_message deploy_message{
-        .kind = EVMC_CALL,
+        .kind = message.depth > 0 ? message.kind : EVMC_CALL,
```

Comment:
The last change to this line made in commit https://github.com/torquem-ch/silkworm/pull/1226/commits/524ae96c1f8418c467447b702ea6aad49fea6155 is wrong:

- here you must go back to `.kind = message.kind,`
- you need to move this logic into the unit tests where checks fail (https://app.circleci.com/pipelines/github/torquem-ch/silkworm/7821/workflows/28458b9e-bff0-42d2-9e0d-cad72b400903/jobs/34593)